logstash-core/lib/logstash/instrument/periodic_poller/jvm.rb

```diff
@@ -22,31 +40,46 @@ def initialize(metric, options = {})
     end
 
     def collect
-      raw = JRMonitor.memory.generate      
+      raw = JRMonitor.memory.generate
       collect_heap_metrics(raw)
       collect_non_heap_metrics(raw)
       collect_pools_metrics(raw)
       collect_threads_metrics
       collect_process_metrics
+      collect_gc_stats
     end
 
     private
 
-    def collect_threads_metrics      
+    def collect_gc_stats
+      garbage_collectors = ManagementFactory.getGarbageCollectorMXBeans()
+
+      garbage_collectors.each do |collector|
+        name = GarbageCollectorName.get(collector.getName())
+        if name.nil?
+          logger.error("Unkown garbage collector name", :name => name)
```

Comment:
I can raise an exception here and log it at the periodic poller thread level.
I dont think should just stop logstash because we can get that information, I believe problem shouldnt stop Logstash processing.